operator/src/main/java/oracle/kubernetes/operator/JobWatcher.java

```diff
@@ -231,14 +243,21 @@ public NextAction apply(Packet packet) {
       return doSuspend(
           (fiber) -> {
             Complete complete =
-                (V1Job job) -> {
+                (V1Job job, boolean isJobFailed) -> {
                   if (!shouldProcessJob(job)) {
                     return;
                   }
                   completeCallbackRegistrations.remove(job.getMetadata().getName());
                   if (didResume.compareAndSet(false, true)) {
                     LOGGER.fine("Job status: " + job.getStatus());
                     packet.put(ProcessingConstants.DOMAIN_INTROSPECTOR_JOB, job);
+                    // Do not proceed to next step such as ReadDomainIntrospectorPodLog if job
+                    // failed due to DeadlineExceeded, as the pod container would likely not
+                    // be available for reading
```

Comment:
Hmm.  It seems like there's a good chance that the timeout could be because the job's scripts themselves were running slowly.  I assume the pod log would still be available in this case?  If so, it'd still be useful to at least echo the contents of the pod log as a 'LOGGER.fine'.

Reply:
unfortunately not as k8s has started terminating the pod. And the readPodLog call that follows would result in "ApiException: Bad Request" error quoted in the PR comment.

Reply:
ah, ok, resolving